packages/flutter_tools/lib/src/hot.dart

```diff
@@ -477,16 +475,16 @@ class HotRunner extends ResidentRunner {
       } else {
         flutterUsage.sendEvent('hot', 'reload');
       }
-    } catch (errorMessage) {
+    } catch (errorMessage, st) {
       reloadStatus.stop(showElapsedTime: true);
-      printError('Hot reload failed:\n$errorMessage');
+      printError('Hot reload failed:\n$errorMessage\n$st');
```

Comment:
You can do a `new Chain.forTrace(st).terse` here to get a more readable stack trace.